packages/ui/src/common/components/page/PreviousPage.tsx

```diff
@@ -8,14 +9,21 @@ import { Arrow } from '../icons'
 interface PreviousPageProps {
   children?: React.ReactNode
   showOverflow?: boolean
+  customLink?: string
 }
-
-export const PreviousPage = ({ children, showOverflow }: PreviousPageProps) => {
+const setPrevHistory = (history: History<unknown>, customLink: string) => {
+  if (history.action === 'POP' && customLink) {
+    history.push(customLink)
+  } else {
+    history.goBack()
+  }
+}
+export const PreviousPage = ({ children, showOverflow, customLink }: PreviousPageProps) => {
   const history = useHistory()
   return (
     <PreviousPageBlock showOverflow={showOverflow}>
       <PreviousPageButtonContainer>
-        <PreviousPageButton onClick={() => history.goBack()} size="small" square>
+        <PreviousPageButton onClick={() => setPrevHistory(history, customLink!)} size="small" square>
```

Comment:
I suggest to fix this by removing non null assertion and marking the customLink as nullable inside the method:
<img width="489" alt="image" src="https://user-images.githubusercontent.com/9252017/208265345-95057128-a7d6-42c9-b525-b6b7676e53e2.png">

<img width="639" alt="image" src="https://user-images.githubusercontent.com/9252017/208265401-611ab581-85cc-462a-957a-0398e8450154.png">